controllers/ptpconfig_controller.go

```diff
@@ -112,7 +127,325 @@ func (r *PtpConfigReconciler) syncPtpConfig(ctx context.Context, ptpConfigList *
+		return nil
+	}
+
+	// 4. Get the linuxptp-daemon DaemonSet
+	daemonSet := &appsv1.DaemonSet{}
+	err := r.Get(ctx, types.NamespacedName{
+		Namespace: names.Namespace,
+		Name:      "linuxptp-daemon",
+	}, daemonSet)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			glog.Info("linuxptp-daemon DaemonSet not found yet - will update on next reconcile")
+			return nil
+		}
+		return fmt.Errorf("failed to get linuxptp-daemon DaemonSet: %v", err)
+	}
+
+	// 5. Compute hash of secret data for change detection
+	secretHash := computeSecretHash(sec)
+	glog.Infof("Computed secret hash for '%s': %s", chosenSecretName, secretHash)
+
+	// 6. Inject the security volume and mounts
+	glog.Infof("Updating linuxptp-daemon DaemonSet with secret '%s' at paths: %v", chosenSecretName, saFilePaths)
+	injectPtpSecurityVolume(daemonSet, chosenSecretName, chosenSecretKey, chosenSecretKeys, saFilePaths, secretHash)
+
+	// 7. Update the DaemonSet
+	if err := r.Update(ctx, daemonSet); err != nil {
+		return fmt.Errorf("failed to update linuxptp-daemon DaemonSet: %v", err)
+	}
+
+	glog.Info("Successfully updated linuxptp-daemon DaemonSet with security mounts")
+	return nil
+}
+
+// computeSecretHash generates a SHA256 hash of the secret data for change detection
+func computeSecretHash(secret *corev1.Secret) string {
+	h := sha256.New()
+
+	// Sort keys for deterministic hash
+	keys := make([]string, 0, len(secret.Data))
+	for k := range secret.Data {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	// Hash each key-value pair
+	for _, k := range keys {
+		h.Write([]byte(k))
+		h.Write([]byte(":"))
+		h.Write(secret.Data[k])
+		h.Write([]byte(";"))
+	}
+
+	return fmt.Sprintf("%x", h.Sum(nil))
+}
+
+// extractSaFileFromPtp4lConf scans the [global] section for a line starting with 'sa_file'
+// and returns the remainder of the line as the path.
+func extractSaFileFromPtp4lConf(conf string) (string, bool) {
+	inGlobal := false
+	sectionHeader := regexp.MustCompile(`^\s*\[[^\]]+\]\s*$`)
+	for _, raw := range strings.Split(conf, "\n") {
+		line := raw
+		// strip inline comments '#' and ';'
+		if idx := strings.IndexAny(line, "#;"); idx >= 0 {
+			line = line[:idx]
+		}
+		line = strings.TrimSpace(line)
+		if line == "" {
+			continue
+		}
+		if sectionHeader.MatchString(line) {
+			if strings.EqualFold(strings.TrimSpace(line), "[global]") {
+				inGlobal = true
+				continue
+			}
+			if inGlobal {
+				// leaving global section
+				break
+			}
+			continue
+		}
+		if !inGlobal {
+			continue
+		}
+		// check for sa_file directive
+		if strings.HasPrefix(line, "sa_file") {
+			parts := strings.Fields(line)
+			if len(parts) >= 2 {
+				return parts[1], true
+			}
+		}
+	}
+	return "", false
+}
+
+// injectPtpSecurityVolume adds a secret volume and mounts to the DaemonSet
+func injectPtpSecurityVolume(ds *appsv1.DaemonSet, secretName string, secretKey string, secretKeys map[string]struct{}, saFilePaths []string, secretHash string) {
```

Comment:
The injectPtpSecurityVolume function only needs: ds *appsv1.DaemonSet, secretName string, saFilePath string, secretHash string. 
This is enough to insert one volume corresponding to a ptpconfig->profile->sa_file mount (corresponding to also one secret)